### DISCUSSION TITLE
[MongoDB] Unique constraint failed on the constraint: `User_username_key`

### DISCUSSION THREAD

**Discussion by bn-piar** [prisma/prisma#16937] [Q&A] [answered]
Hi. I have a `User` model: 

```prisma
model User {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  email     String   @unique
  password  String
  name      String?
  role      UserRole @default(Client)
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
}

enum UserRole {
  Admin
  Client
  Sale
}
```
In DB a `User` document may look like this
```json
{
  "_id": {
    "$oid": "63a28da629d26656f840a340"
  },
  "email": "Kenya39@yahoo.com",
  "password": "2FHe2yUv5VSjYEd",
  "name": "Karen Cummings",
  "role": "Admin",
  "createdAt": {
    "$date": {
      "$numberLong": "1671597478747"
    }
  },
  "updatedAt": {
    "$date": {
      "$numberLong": "1671597478747"
    }
  }
}
```

I can create only one new user. After that creation fails with error `Unique constraint failed on the constraint: 'User_username_key'`.

Payload to create the first user and response:
```json
// payload
{
    "name": "Karen Cummings",
    "email": "Kenya39@yahoo.com",
    "password": "2FHe2yUv5VSjYEd",
    "role": "Admin"
}
// response
{
    "id": "63a28da629d26656f840a340",
    "email": "Kenya39@yahoo.com",
    "emailHash": null,
    "password": "2FHe2yUv5VSjYEd",
    "name": "Karen Cummings",
    "role": "Admin",
    "createdAt": "2022-12-21T04:37:58.747Z",
    "updatedAt": "2022-12-21T04:37:58.747Z"
}
```
Payload to create the second user and response:
```json
//payload
{
    "name": "Glenda Runte",
    "email": "Malachi_Mayer@yahoo.com",
    "password": "rU9R1GKyOVhCrR5",
    "role": "Client"
}
// response error
{
    "code": "P2002",
    "clientVersion": "4.8.0",
    "meta": {
        "target": "User_username_key"
    }
}
```
That's how I create a user:
```typescript
function createUser(user: CreateUserDto) {
  return prisma.user.create({
    data: user,
  });
}
...
const { name, email, password, role }: CreateUserDto = req.body;
try {
  const user = await createUser({
    name,
    email,
    password,
    role,
  });
  res.json(user);
}
```

If I rename model to something else, like `dbUser`, everything works just fine. 

Does anyone else experience this?

**Comment by nurul3101** [comment, marked as answer]
Hey @bn-piar 👋 

Did your model had a field called username before? I can see that the constraint is failing for a `username` field but it doesn't exist in your model. Had you invoked `npx prisma db push` after making changes to your schema file?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
